dll/win32/shell32/dialogs/filedefext.cpp

```diff
@@ -1304,25 +1371,19 @@ CFileDefExt::_CountFolderAndFilesThreadProc(LPVOID lpParameter)
 }
 
 BOOL
-CFileDefExt::CountFolderAndFiles(HWND hwndDlg, LPWSTR pwszBuf, UINT cchBufMax, DWORD *ticks)
+CFileDefExt::CountFolderAndFiles(HWND hwndDlg, LPWSTR pwszBuf, DWORD *ticks)
 {
-    /* Find filename position */
-    UINT cchBuf = wcslen(pwszBuf);
-    WCHAR *pwszFilename = pwszBuf + cchBuf;
-    size_t cchFilenameMax = cchBufMax - cchBuf;
-    if (!cchFilenameMax)
-        return FALSE;
-    *(pwszFilename++) = '\\';
-    --cchFilenameMax;
-
-    /* Find all files, FIXME: shouldn't be "*"? */
-    StringCchCopyW(pwszFilename, cchFilenameMax, L"*");
+    CString sBuf = pwszBuf;
+    sBuf += L"\\" ;
+    CString sSearch = sBuf;
+    sSearch += L"*" ;
+    CString sFileName;
 
     WIN32_FIND_DATAW wfd;
-    HANDLE hFind = FindFirstFileW(pwszBuf, &wfd);
+    HANDLE hFind = FindFirstFileW(sSearch.GetBuffer(), &wfd);
     if (hFind == INVALID_HANDLE_VALUE)
     {
-        ERR("FindFirstFileW %ls failed\n", pwszBuf);
+        ERR("FindFirstFileW %ls failed\n", sSearch.GetBuffer());
```

Comment:
`GetString()` is required here, this one cannot be automatically casted.

Reply:
ok